arcgis-ios-sdk-samples/Layers/Query with CQL filters/QueryWithCQLFiltersSettingsViewController.swift

```diff
@@ -0,0 +1,144 @@
+        let queryParameters = AGSQueryParameters()
+        queryParameters.whereClause = selectedWhereClause
+        queryParameters.maxFeatures = maxFeatures
+        if dateFilterSwitch.isOn {
+            queryParameters.timeExtent = AGSTimeExtent(startTime: startDatePicker.date, endTime: endDatePicker.date)
+        }
+        return queryParameters
+    }
+    
+    func showWhereClausePicker() {
+        let selectedIndex = sampleWhereClauses.firstIndex { $0 == selectedWhereClause }
+        let optionsViewController = OptionsTableViewController(labels: sampleWhereClauses, selectedIndex: selectedIndex) { newIndex in
+            let whereClause = self.sampleWhereClauses[newIndex]
+            self.selectedWhereClause = whereClause
+            self.whereClauseCell.detailTextLabel?.text = whereClause
+            self.navigationController?.popViewController(animated: true)
+        }
+        optionsViewController.title = "Where Clause"
+        show(optionsViewController, sender: self)
+    }
+    
+    /// Set the date picker limits to maintain a valid state.
+    func updateDatePickerLimits() {
+        // Keep the start date less than the end date.
+        startDatePicker.maximumDate = Calendar.current.date(byAdding: .day, value: -1, to: endDatePicker.date)
+        // Keep the end date greater than the start date.
+        endDatePicker.minimumDate = Calendar.current.date(byAdding: .day, value: 1, to: startDatePicker.date)
+    }
+    
+    @IBAction func datePickerAction(_ sender: Any) {
+        updateDatePickerLimits()
+    }
+    
+    @IBAction func cancelAction(_ sender: UIBarButtonItem) {
+        dismiss(animated: true)
+    }
+    
+    @IBAction func applyAction(_ sender: UIBarButtonItem) {
+        dismiss(animated: true)
+        delegate?.settingsViewController(self, didCreate: makeQueryParameters())
+    }
+    
+    @IBAction func textFieldEditingDidEnd(_ sender: UITextField) {
+        if let text = sender.text,
+           let value = self.numberFormatter.number(from: text),
+           value.intValue > 0 {
+            maxFeatures = value.intValue
+        } else {
+            // Reset to default value if the input is invalid.
+            maxFeatures = 1000
+            sender.text = "1000"
+        }
+    }
+    
+    @IBAction func switchValueChanged(_ sender: UISwitch) {
+        startDatePicker.isEnabled = sender.isOn
+        endDatePicker.isEnabled = sender.isOn
+    }
+    
+    // MARK: UITableViewController
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        let cell = tableView.cellForRow(at: indexPath)
+        switch cell {
+        case whereClauseCell:
+            showWhereClausePicker()
+        default:
+            return
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.keyboardDismissMode = .onDrag
```

Comment:
This can be set through Interface Builder.

![image](https://user-images.githubusercontent.com/2257493/122111880-18a31080-cdd5-11eb-82e4-dd8480359500.png)